yaksh/models.py

```diff
@@ -1164,13 +1180,27 @@ class AnswerPaper(models.Model):
         default='inprogress'
     )
 
+    # set question order
+    questions_order = models.CharField(max_length=255, blank=True, default='')
+
     objects = AnswerPaperManager()
 
     def current_question(self):
         """Returns the current active question to display."""
         if self.questions_unanswered.all():
-            return self.questions_unanswered.all()[0]
-        return self.questions.all()[0]
+            cur_question = self.get_current_question(
+                self.questions_unanswered.all())
```

Comment:
You are making two queries here so save the unanswered_questions here and then use that.